### PR TITLE
fix build on mac and add instructions

### PR DESCRIPTION
**Pull request by 3nids**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -80,3 +80,16 @@ cp config.pri.default config.pri
...
+
+ #### On Mac
+
+Building on a Mac requires some adjustments:
+
+1. In Qt Creator > Projects > Run >
+* Check "Use debug version of frameworks"
+* In Run Environment, edit following variables (replace `_QGIS_INSTALL_` by QGIS installation path):
+  * `DYLD_FRAMEWORK_PATH` add `:_QGIS_INSTALL_/QGIS.app/Contents/Frameworks`
+  * `DYLD_LIBRARY_PATH` add `:_QGIS_INSTALL_/QGIS.app/Contents/Frameworks/qgis_core.framework/Versions/Current`
```
Comment (m-kuhn) [inline README.md]: Might be possible to use homebrew paths by default?

**Comment by m-kuhn** [comment]
Uploaded test apk to https://opengis.ch/download/qfield/ci-builds/qfield-dev-237-6a2ca7f274b1a5fbaef2cef3cfc046bafecaa07a.apk

File: qgis.pri [hunk 2]
```diff
@@ -25,10 +35,15 @@ android {
...
 }
 
-DEFINES += "QGIS_PLUGIN_DIR=\\\"$$QGIS_INSTALL_PATH$$/lib/qgis/plugins/\\\"" \
-           "QGIS_INSTALL_DIR=\\\"$$QGIS_INSTALL_PATH$$\\\""
-
-QMAKE_RPATHDIR += $${QGIS_INSTALL_PATH}/lib/
+macx {
+  DEFINES += "QGIS_PLUGIN_DIR=\\\"$$QGIS_INSTALL_PATH$$/QGIS.app/Contents/PlugIns/qgis\\\"" \
+             "QGIS_INSTALL_DIR=\\\"$$QGIS_INSTALL_PATH$$\\\""
+  QMAKE_RPATHDIR += $${QGIS_INSTALL_PATH}//Contents/MacOS/lib/
```
Comment (m-kuhn) [inline qgis.pri]: double-/ on purpose?

File: qgis.pri [hunk 3]
```diff
@@ -5,10 +5,20 @@
...
+    QMAKE_LFLAGS += -F$${QGIS_INSTALL_PATH}/QGIS.app/Contents/Frameworks/
+    LIBS += -F$${QGIS_INSTALL_PATH}/QGIS.app/Contents/Frameworks/
+    LIBS += -framework qgis_core
 
+    INCLUDEPATH += $${QGIS_INSTALL_PATH}/QGIS.app/Contents/Frameworks/qgis_core.framework/Versions/Current/Headers
+    INCLUDEPATH += $${GEOS_INCLUDE_PATH}
+  }
+  !macx {
+    INCLUDEPATH += $${QGIS_INSTALL_PATH}/include/qgis
+    INCLUDEPATH += $${GEOS_INCLUDE_PATH}
```
Comment (m-kuhn) [inline qgis.pri]: That one is new. On purpose?

File: qgis.pri [hunk 4]
```diff
@@ -5,10 +5,20 @@
 include( paths.pri )
 
 !android {
-  INCLUDEPATH += $${QGIS_INSTALL_PATH}/include/qgis
-  LIBS += $${QGIS_INSTALL_PATH}/lib/libqgis_core.so
-}
+  macx {
+    QMAKE_LFLAGS += -F$${QGIS_INSTALL_PATH}/QGIS.app/Contents/Frameworks/
+    LIBS += -F$${QGIS_INSTALL_PATH}/QGIS.app/Contents/Frameworks/
```
Comment (m-kuhn) [inline qgis.pri]: Do we set both, `LIBS` and `QMAKE_LFLAGS`?

**Comment by m-kuhn** [comment]
Nice one, just some minor comments

**Comment by 3nids** [comment]
comments addressed!

File: paths.pri [hunk 5]
```diff
@@ -14,6 +14,6 @@ android {
   QT_LIBS_DIR = $$dirname(QMAKE_QMAKE)/../lib
 
 }
-!android {
+!android:!macx {
```
Comment (m-kuhn) [inline paths.pri]: So `:` equals to `&&`?

**Pull request merged**